scripts/us_cdc/natality/clean_cdc_data.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A script to clean CDC Wonder Natality files."""
+
+import os
+import re
+
+from absl import app
+from absl import flags
+
+flags.DEFINE_string('input_path', None, 'Path to directory with files to clean')
+flags.DEFINE_string(
+    'output_path', None,
+    'Path to output directory. If empty, files are overwritten')
+
+_FLAGS = flags.FLAGS
+
+# Regex to identify "Notes" in the header file
+_HEADER_REGEX = re.compile(r'"Notes"')
+
+# Regex to identify tabs at the start of the line
+_STARTING_TAB_REGEX = re.compile(r'^\t+')
+
+# Regex to identify tab delimiters
+_TAB_DELIMITER_REGEX = re.compile(r'\t+')
+
+
+def clean_file(f_handle, w_handle=None):
+    """Clean CDC Data file.
+
+    Removes '"Notes"' from header.
+    Removes lines that start with '"Total"'.
+    All lines after first occurance of '"---"' are removed.
+
+    Args:
+        f_handle: File handle of file to clean.
+        w_handle: File to write to. If None, attempts to overwrite f_handle.
+    """
+    lines = f_handle.readlines()
+    if not w_handle:
+        w_handle = f_handle
+        w_handle.truncate(0)
+
+    for line in lines:
+        if line.startswith('"Notes"'):  # header line
+            # Removing "Notes"
+            cleaned_header = _HEADER_REGEX.sub('', line)
+
+            # Tabs at start of the line are removed
+            cleaned_header = _STARTING_TAB_REGEX.sub('', cleaned_header)
+
+            # Tab delimiter replaced with ,
+            cleaned_header = _TAB_DELIMITER_REGEX.sub(',', cleaned_header)
+
+            w_handle.write(cleaned_header)
+
+        elif not line.startswith('"Total"'):  # Line should not start with total
```

Comment:
line.startswith("----") should be above this condition?

Reply:
Done.

Reply:
The ordering of the if elif ladder has changed. The code works as expected. The typical structure of a text file in the source data is as follows:

`Notes ...` (Header line with column names)
`Alabama ...` (Data)
`California ...`
...
`Total ...`
`---` (Start of meta data)

The header block is executed for the first line. The `line.startswith('"Total"')` block is executed for all the data lines. When the meta data line is encountered the script breaks out of the loop and stops writing to the cleaned file.